packages/tasit-sdk/src/Decentraland.forkedRopsten.test.js

```diff
@@ -0,0 +1,221 @@
+
+const { ONE, TEN, BILLION } = constants;
+
+const ROPSTEN_NETWORK_ID = 3;
+
+describe("Decentraland tasit app test cases (ropsten)", () => {
+  let ownerWallet;
+  let ephemeralWallet;
+  let manaContract;
+  let landContract;
+  let estateContract;
+  let marketplaceContract;
+  let landForSale;
+  let estateForSale;
+  let snapshotId;
+  let provider;
+
+  before("", async () => {
+    ConfigLoader.setConfig(config);
+
+    provider = ProviderFactory.getProvider();
+
+    const network = await provider.getNetwork();
+    const { chainId } = network;
+    expect(chainId, "The target network isn't ropsten.").to.equal(
+      ROPSTEN_NETWORK_ID
+    );
+
+    manaContract = new Mana(MANA_ADDRESS);
+    landContract = new Land(LAND_ADDRESS);
+    estateContract = new Estate(ESTATE_ADDRESS);
+    marketplaceContract = new DecentralandMarketplace(MARKETPLACE_ADDRESS);
+
+    const decentralandUtils = new DecentralandUtils();
+    const { getOpenSellOrders } = decentralandUtils;
+
+    const fromBlock = 0;
+    const openSellOrders = await getOpenSellOrders(fromBlock);
+
+    // Note: The exact amount of land isn't predictable since we are forking from the latest block
+    expect(openSellOrders).to.not.be.empty;
+
+    // Pick a land and an estate open sell orders
+    for (let sellOrder of openSellOrders) {
+      const { values: order } = sellOrder;
+      const { nftAddress, expiresAt } = order;
+
+      if (landForSale && estateForSale) break;
+
+      const isLand = addressesAreEqual(nftAddress, LAND_ADDRESS);
+      const isEstate = addressesAreEqual(nftAddress, ESTATE_ADDRESS);
+      const expired = Number(expiresAt) < Date.now();
+
+      // All lands are expired
+      if (isLand) landForSale = order;
+      if (isEstate && !expired) estateForSale = order;
+
+      if (!isLand && !isEstate)
+        expect(
+          false,
+          "All land for sale should be a land or an estate NFT"
+        ).to.equal(true);
+    }
+
+    expect(estateForSale).to.not.be.an("undefined");
+    expect(landForSale).to.not.be.an("undefined");
+  });
+
+  beforeEach(
+    "buyer approve marketplace contract to transfer tokens on their behalf",
+    async () => {
+      snapshotId = await createSnapshot(provider);
+
+      ({ ownerWallet, ephemeralWallet } = setupWallets());
+      expect(ownerWallet.address).to.have.lengthOf(42);
+      expect(ephemeralWallet.address).to.have.lengthOf(42);
+
+      await etherFaucet(provider, ownerWallet, ephemeralWallet, ONE);
+
+      await confirmBalances(manaContract, [ephemeralWallet.address], [0]);
+      await ropstenManaFaucet(provider, ownerWallet, ephemeralWallet, BILLION);
+      await confirmBalances(manaContract, [ephemeralWallet.address], [BILLION]);
+
+      manaContract.setWallet(ephemeralWallet);
+      const approvalAction = manaContract.approve(
+        MARKETPLACE_ADDRESS,
+        BILLION,
+        gasParams
+      );
+      await approvalAction.waitForNonceToUpdate();
```

Comment:
Are you talking about `gasParams`?

Reply:
No, about `waitForNonceToUpdate`. In theory, our build action, send, and wait for success event flow is supposed to be a better fit for developing front ends given the delay from the blockchain. So it's interesting that it's seemed more convenient for you to use the traditional more-ethers.js style API in the tests. Maybe it's because the test blockchain is instant, so the delay that makes pubsub warranted doesn't come into play here?

Reply:
Yes, test suites expect a "sync behavior" from the test cases.

Reply:
Could I resolve this one?

Reply:
Hmmm. I mean, we're still using an async `beforeEach` function here, and `waitForNonceToUpdate` is async. So I don't see how that's any more "sync-like" than sending an action and then listening for the result.

Maybe we want to explicitly call `done()` for a non-async test in the listener for the expected event rather than using `async` like it mentioned here https://github.com/mochajs/mocha/issues/2988#issuecomment-327200194

It's quite possible I'm missing something here that would make this not work, though.

Reply:
Hmm, I think you're right, I also think that we can use `done()` for a sync test case if we want to.
I've used work `sync` wrongly. What I meant with  "sync behavior" was that test suite looks clear with sequential / no-thread dispatching code style.

I think that the main question here is if we are keeping the current style of will be better try to use `done()` to get closer as we expected that the code will be (without `waitForNonceToUpdate`).

Reply:
If there's no thread-dispatching, then how can we write tests for things that involve thread dispatching? I guess if we have one unit test that confirms that a thread is successfully dispatched and a separate test that confirms that the thread-dispatching function is called that would address that.

I agree that that's the main question. And I'm leaning towards thinking that the other style better captures how we expect the library to be used (and that's an important secondary purpose of tests beyond confirming desired functionality)

Reply:
Extracted to: https://github.com/tasitlabs/TasitSDK/issues/219

Reply:
Thanks! Resolving